hexa/workspaces/models.py

```diff
@@ -133,3 +134,141 @@ class WorkspaceMembership(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
     objects = WorkspaceMembershipManager.from_queryset(WorkspaceMembershipQuerySet)()
+
+
+class WorkspaceConnectionQuerySet(BaseQuerySet):
+    def filter_for_user(
+        self, user: typing.Union[AnonymousUser, User]
+    ) -> models.QuerySet:
+        return self._filter_for_user_and_query_object(user, Q(workspace__members=user))
+
+
+class WorkspaceConnectionManager(models.Manager):
+    def create_if_has_perm(
+        self, principal: User, workspace: Workspace, fields=None, **kwargs
+    ):
+        if not principal.has_perm("workspaces.create_workspace_connection", workspace):
+            raise PermissionDenied
+
+        connection = self.create(workspace=workspace, user=principal, **kwargs)
+        if fields is not None:
+            connection.set_fields(principal, fields)
+
+        return connection
+
+
+class WorkspaceConnectionType(models.TextChoices):
+    S3 = "S3", _("S3 Bucket")
+    GCS = "GCS", _("GCS Bucket")
+    POSTGRESQL = "POSTGRESQL", _("PostgreSQL DB")
+    CUSTOM = "CUSTOM", _("Custom")
+
+
+class WorkspaceConnection(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    workspace = models.ForeignKey(
+        Workspace, on_delete=models.CASCADE, related_name="connections"
+    )
+    user = models.ForeignKey(
+        "user_management.User",
+        on_delete=models.CASCADE,
+    )
+    name = models.CharField(max_length=255, null=False)
+    _slug = models.CharField(
```

Comment:
I would suggest to always explicitly store the slug instead of having a `@property` - this would prevent code from breaking should the name change at some point.

The slug could be set to the slugified name either in the frontend or in the backend but saved in an explicit `slug` field

Reply:
On create, take the name and slugify it. On update, do nothing if slug is not passed.